### PR TITLE
Adding asserts for hanging cases in ND tilize/untilize support

### PR DESCRIPTION
**Pull request by nardoTT**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/16141)

### Problem description
For ND support, tilize/untilize currently hang for shapes with values different than 1 for the dimensions greater than 4.

### What's changed
Add FATALs to avoid these hangs

### Checklist
- [x] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/12416274632
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by kmabeeTT** [comment]
Thanks for PR - I'm going to be cherry picking this as soon as it gets merged to unblock tt-metal uplift into tt-mlir.

File: ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp:26]: Can you please give it a more clear name?
This function is something like validate_nd_support right?

**Review by ayerofieiev-tt — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp:30]: This can be a TT_FATAL and you should print out i and initial_shape[i] in the message.

**Review by bbradelTT — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/core/work_split/work_split_tilize.hpp
Comment (sjameelTT) [inline ttnn/cpp/ttnn/operations/core/work_split/work_split_tilize.hpp:189]: delete this

**Pull request merged**
